pkg/compact/compact.go

```diff
@@ -157,6 +160,42 @@ func (c *Syncer) SyncMetas(ctx context.Context) error {
 }
 
 func (c *Syncer) syncMetas(ctx context.Context) error {
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
did not check but I guess running `wg.Wait()` twice is fine?

Reply:
Yeah it's fine.  `wg.Wait()` just blocks until the counter is zero, so no problem with doing it multiple times.